plugins/network-elements/nsx/src/main/java/org/apache/cloudstack/service/NsxGuestNetworkGuru.java

```diff
@@ -265,20 +275,22 @@ public void commitMigration(NicProfile nic, Network network, VirtualMachineProfi
     }
 
     private void createNsxSegment(NetworkVO networkVO, DataCenter zone) {
-            String vpcName = null;
-        if (nonNull(networkVO.getVpcId())) {
-            VpcVO vpc = _vpcDao.findById(networkVO.getVpcId());
-            if (isNull(vpc)) {
-                throw new CloudRuntimeException(String.format("Failed to find VPC network with id: %s", networkVO.getVpcId()));
-            }
-            vpcName = vpc.getName();
+        VpcVO vpc = _vpcDao.findById(networkVO.getVpcId());
+        if (isNull(vpc)) {
+            throw new CloudRuntimeException(String.format("Failed to find VPC network with id: %s", networkVO.getVpcId()));
```

Comment:
This refactor may not be required. The idea of it being the way it was previously written was that isolated networks could also be created as segments, and they would be connected to tier0

Reply:
Thanks, reverted that part